activemodel/lib/active_model/type/float.rb

```diff
@@ -7,6 +7,10 @@ def type
         :float
       end
 
+      def type_cast_for_schema(value)
+        value.inspect.gsub('Infinity', '::Float::INFINITY')
```

Comment:
Is there any case here other than infinity and negative infinity? Could this be more explicit and be

```ruby
case value
  when ::Float::INFINITY then "::Float::INFINITY"
  when -::Float::INFINITY then "-::Float::INFINITY"
  else super
end
```